main.tf

```diff
@@ -89,6 +89,16 @@ resource "aws_s3_bucket" "bucket" {
       }
     }
   }
+  
+  dynamic "grant" {
+    for_each = var.grants
+    iterator = grant
+    content {
+      id          = lookup(grant.value, "id", null)
+      type        = lookup(grant.value, "type", "CanonicalUser")
+      permissions = lookup(grant.value, "permissions", ["FULL_CONTROL"])
```

Comment:
I don't think the default value should be `FULL_CONTROL` here. It violates the least privilege principle for most users. I think a more sensible default permission would be `READ`.

Reply:
makes sense. changing

Reply:
Thanks. Everything else looks good to me.